test/metabase/channel/impl/email_test.clj

```diff
@@ -20,3 +22,46 @@
                                                  :message      "Test message"})
           (is (=? {:to ["ngoc@metabase.com"]}
                   @sent-message)))))))
+
+(deftest assoc-attachment-booleans-test
+  (testing "assoc-attachment-booleans function"
+    (testing "handles visualizer dashcards by matching on both card_id and dashboard_card_id"
+      (let [;; Create a visualizer dashcard
```

Comment:
nit - just my preference but I don't think these kind of comments are needed when they basically just echo what the code says and don't add extra context